File: Backend/Iteramico/DomainRepository/Models/Reminder.cs
namespace DomainRepository.Models
{
#nullable disable
    public class Reminder
    {
        public int Id { get; set; }
        public string Note { get; set; }
        public DateTime Deadline { get; set; }

        public int CreatorId { get; set; }
        public virtual User Creator { get; set; }

        public int ForId { get; set; }
        public virtual User For { get; set; }

        public int JourneyId { get; set; }
        public virtual Journey Journey { get; set; }
    }
}

File: frontend/lib/models/reminder.dart
import 'package:frontend/models/user.dart';

class Reminder {
  final int id;
  final String note;
  final DateTime deadline;
  final User refersTo;
  final User creator;

  Reminder(
      {required this.id,
      required this.note,
      required this.deadline,
      required this.refersTo,
      required this.creator});

  factory Reminder.fromJson(Map<String, dynamic> json) {
    return Reminder(
        id: json['id'],
        note: json['note'],
        deadline: DateTime.parse(json['deadline']),
        creator: User.fromJson(json['creator']),
        refersTo: User.fromJson(json['for']));
  }
}
